D2 only; adapt to ocean, swarm, turtle new majors

gavin-norman-sociomantic opened the pull request.

Commit:
Commit result of d1to2fix

Commit:
Update to new major of ocean, swarm, turtle

* submodules/ocean v4.0.11(b85bb255)...v5.0.4(5e47d4df) (150 commits)
  > Merge tag v4.6.2 into v5.0.x
  > Explicitly handle `void*` types in `ReduceType`
  > Add sanity tests of instantiations of `ReduceType`
  > Merge tag v4.6.1 into v5.0.x
  > Test with upstream DMD 2.085
  (...)

* submodules/ocean v4.0.11(b85bb255)...v5.0.4(5e47d4df) (150 commits)
  > Merge tag v4.6.2 into v5.0.x
  > Explicitly handle `void*` types in `ReduceType`
  > Add sanity tests of instantiations of `ReduceType`
  > Merge tag v4.6.1 into v5.0.x
  > Test with upstream DMD 2.085
  (...)

* submodules/swarm v5.0.20(186de47)...v6.0.0-rc.3(fa7475d) (118 commits)
  > Add static checks for members of `Request` type in `ConnectionHandler`
  > Remove deprecated static ifs
  > Explicitly exclude pointers in `StructPacker`
  > Remove unused module `swarm.neo.util.VoidBufferAsArrayOf`
  > Use `ocean.util.container.VoidBufferAsArrayOf`
  (...)

* submodules/turtle v9.0.2(c486df9)...v10.0.3(c10a254) (26 commits)
  > Merge tag v9.1.5 into v10.0.x
  > Switch to more recent upstream DMD in CI
  > Upgrade ocean to get rid of deprecations
  > Release note
  > Adapt CI to be D2 only
  (...)

Commit:
Fixups for new major submodules

Commit:
Mark repo as D2-only

Commit:
Update build directives to be D2-only

Commit:
Remove D1 tests from CI

gavin-norman-sociomantic commented:
Current progress investigating the test failures:
* The Put request fails with a `node_disconnected` notification.
* I've not tracked down why this happens.

tiyash-basu-sociomantic commented:
The following change makes the tests pass:
```diff
diff --git a/src/dlsproto/node/neo/request/GetRange.d b/src/dlsproto/node/neo/request/GetRange.d
index 08f8175..b4a56cc 100644
--- a/src/dlsproto/node/neo/request/GetRange.d
+++ b/src/dlsproto/node/neo/request/GetRange.d
@@ -42,6 +42,7 @@ public abstract class GetRangeProtocol_v2: IRequest
     import swarm.util.RecordBatcher;
     import swarm.neo.node.RequestOnConn;
     import dlsproto.common.GetRange;
+    import dlsproto.node.neo.request.core.IRequestResources;
     public import dlsproto.client.request.GetRange: Filter;
 
     import core.stdc.time;
@@ -147,6 +148,9 @@ public abstract class GetRangeProtocol_v2: IRequest
     void handle ( RequestOnConn connection, Object resources,
         Const!(void)[] init_payload )
     {
+        this.ed = connection.event_dispatcher;
+        auto rq_resources = cast(IRequestResources)resources;
+
         cstring channel_name;
         cstring filter_string;
         time_t low, high;
@@ -190,11 +194,11 @@ public abstract class GetRangeProtocol_v2: IRequest
                 }
             );
 
-            this.value_buffer = this.resources.getVoidBuffer();
-            this.batch_buffer = this.resources.getVoidBuffer();
-            this.compressed_batch = this.resources.getVoidBuffer();
-            this.saved_exception = this.resources.getException();
-            this.lzo = this.resources.getLzo();
+            this.value_buffer = rq_resources.getVoidBuffer();
+            this.batch_buffer = rq_resources.getVoidBuffer();
+            this.compressed_batch = rq_resources.getVoidBuffer();
+            this.saved_exception = rq_resources.getException();
+            this.lzo = rq_resources.getLzo();
 
             try
             {
diff --git a/src/dlsproto/node/neo/request/Put.d b/src/dlsproto/node/neo/request/Put.d
index 74ff53c..a3fa50e 100644
--- a/src/dlsproto/node/neo/request/Put.d
+++ b/src/dlsproto/node/neo/request/Put.d
@@ -52,10 +52,12 @@ public abstract class PutProtocol_v1: IRequest
     void handle ( RequestOnConn connection, Object resources,
         Const!(void)[] init_payload )
     {
+        auto ed = connection.event_dispatcher;
+
         cstring channel;
         time_t timestamp;
         Const!(char)[] value;
-        this.ed.message_parser.parseBody(init_payload, channel, timestamp, value);
+        ed.message_parser.parseBody(init_payload, channel, timestamp, value);
 
         // Store the extracted data in StorageEngine
         RequestStatusCode response;
@@ -66,7 +68,7 @@ public abstract class PutProtocol_v1: IRequest
             response = RequestStatusCode.Error;
 
         // Send the response code.
-        this.ed.send(
+        ed.send(
             ( RequestOnConnBase.EventDispatcher.Payload payload )
             {
                 payload.addCopy(response);
```

The `IRequestHandlerRequestCore.initialise` method takes care of initialising the needed member variables here, but that method doesn't seem to be needed anymore.

tiyash-basu-sociomantic commented on src/fakedls/neo/request/GetRange.d:
```diff
@@ -41,6 +41,12 @@ public class GetRangeImpl_v2: GetRangeProtocol_v2
+    static immutable Command command = Command(RequestCode.GetRange, 2);
+
+    /// Request name for stats tracking. Required by ConnectionHandler.
+    static immutable istring name = "GetRange";
```
The requests derive from `IRequest`, and will always need `addHandler`, which, in turn, needs the `name` and the `command` members from the request class. However, the current definition of `IRequest` does not enforce having these members.
Do you think that it will be useful to add `getName` and `getCommand` methods to the `IRequest` interface? At this point, it will be trivial to add this change in swarm and *proto.

gavin-norman-sociomantic commented:
Updated.

gavin-norman-sociomantic commented:
Updated to use swarm v6.0.0-rc.3.

Commit:
Add needed static params to fakedls request handlers

`timing` and `scheduled_for_removal` params are needed for the request
handlers to compile.

Commit:
Add tests with dmd v2.086.* to CI

gavin-norman-sociomantic commented:
Updated with extra commits from https://github.com/tiyash-basu-sociomantic/dlsproto/commits/new-major.

tiyash-basu-sociomantic committed:
Change `(&this)` to `this`

These were introduced by d1to2fix, and were not needed.

tiyash-basu-sociomantic reviewed (approved):
LGTM :+1:

matthias-wende-sociomantic commented on .D2-ready:
```diff
@@ -0,0 +1 @@
+ONLY
```
Can you please add the newline character?

matthias-wende-sociomantic commented on Config.mak:
```diff
@@ -0,0 +1 @@
+DVER := 2
```
Can you please add the newline character?

matthias-wende-sociomantic reviewed (approved):
I'll merge as soon as the newlines are added.

gavin-norman-sociomantic commented:
Newlined.
